lib/fog/orchestration/openstack/requests/create_stack.rb

```diff
@@ -28,6 +31,19 @@ def create_stack(arg1, arg2 = nil)
             }.merge(arg2.nil? ? {} : arg2)
           end
 
+          # Templates should always:
+          #  - be strings
+          #  - contain URI references instead of relative paths.
+          # Passing :template_url may not work well with `get_files` and remote `type`:
+          #  the python client implementation in shade retrieves from :template_uri
+          #  and replaces it with :template.
+          #  see https://github.com/openstack-infra/shade/blob/master/shade/openstackcloud.py#L1201
+          #  see https://developer.openstack.org/api-ref/orchestration/v1/index.html#create-stack
+          file_resolver = Fog::Orchestration::Util::RecursiveHotFileLoader.new(options[:template] || options[:template_url], options[:files])
+          files = file_resolver.get_files
+          options[:template] = file_resolver.template
```

Comment:
According to your logic `file_resolver.template` returns a Hash. I assume `options[:template]` expects a string, i.e., the content of the template.